.github/workflows/linter.yml

```diff
@@ -0,0 +1,47 @@
+name: Linter check
+
+env:
+  HUSKY: 0
+
+on:
+  pull_request:
+    branches: ["**"]
+  workflow_dispatch:
+
+jobs:
+  linter:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js 22
+        uses: actions/setup-node@v4
+        with:
+          node-version: 22
+          cache: "yarn"
+
+      - name: Install dependencies
+        run: |
+          corepack enable
+          yarn install
```

Comment:
We don't want to install **all** dependencies here, but only the dev deps needed to perform the checks:
- cspell,
- eslint,
- knip,
- prettier

Reply:
Good point, but separately they are best not put up, for a reason:
1) You will have to install not only these packages, but also additional packages: `typescript-eslint`;
2) It will be necessary to keep track of versions additionally;
3) To make CI work correctly, you need to run codegen (and this requires a package);
4) If you make a common cache for `node_modules` (which you should do), the installation will take seconds.